SCRUM-9 Add Basic Post CRUD Functions

krwro opened the pull request:
Added CRUD endpoints, models, exceptions, and services 
Added Localstack and AWS Dynamo integration
Added necessary dependencies
Added basic documentation
Added basic integration tests
Updated README.md

krwro committed:
SCRUM-9 Add Basic Post CRUD Functions

Added CRUD endpoints, models, exceptions, and services
Added Localstack and AWS Dynamo integration
Added necessary dependencies
Added basic documentation
Added basic integration tests
Updated README.md

ovanov commented on src/main/java/ch/nexusnet/postmanager/aws/dynamodb/model/table/DynamoDBTableInitializer.java:
```diff
@@ -0,0 +1,40 @@
+        this.amazonDynamoDB = amazonDynamoDB;
+    }
+
+    @PostConstruct
+    public void init() {
+        CreateTableRequest tableRequest = dynamoDBMapper
+                .generateCreateTableRequest(DynamoDBPost.class)
+                .withProvisionedThroughput(new ProvisionedThroughput(1L, 1L));
+
+        try {
+            DescribeTableRequest describeTableRequest = new DescribeTableRequest()
+                    .withTableName("Posts");
+            DescribeTableResult describeTableResult = amazonDynamoDB.describeTable(describeTableRequest);
+
+            System.out.println("Table already exists. Table status: " + describeTableResult.getTable().getTableStatus());
+        } catch (ResourceNotFoundException e) {
+            amazonDynamoDB.createTable(tableRequest);
+            System.out.println("Created DynamoDB table: Posts");
+        }
+    }
```
I'm in favour of a code-based table creation, but it can be quite dangerous for our production. Maybe a script would be more appropriate for our approach. Let's discuss this maybe.

krwro replied:
You're right a script would be better

ovanov reviewed (commented):
Everything looks really clean to me, very nice :D
A few unit tests would be nice, but you can also add them later.

r-carissimi reviewed (commented):
Looks good. Well done!

krwro committed:
SCRUM-9 Add Unit Tests and Different Table Init

Added unit tests for post controller and service with TesDataUtil class.

Renamed userId to authorId for clarity and consistency

Added hashtags to PUT

Added init.sh methods for dynamodb table instantiation (Not tested yet)

krwro committed:
SCRUM-9 Fix createPost Integration test

Added todos for ApplicationStartupRunner.

Removed unnecessary comments
